### PR TITLE
#26862: allow using declarationDir with composite option

### LINKED ISSUE #26862: Error using 'declarationDir' with 'composite'
**TypeScript Version:**  3.1.0-dev.20180831

**Search Terms:**

**Code**

tsconfig.json
```ts
{
  "compilerOptions": {
    "composite": true,
    "declarationDir": "./decl"
  }
}
```

**Expected behavior:**

No error because `composite` implicitly enables `declaration`.

**Actual behavior:**

`TS5052: Option 'declarationDir' cannot be specified without specifying option 'declaration'.`

**Related Issues:** #26786

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #26862

### CODE REVIEW

File: src/compiler/program.ts [hunk 1]
```diff
@@ -2532,8 +2532,8 @@ namespace ts {
             }
 
             if (options.declarationDir) {
-                if (!options.declaration) {
-                    createDiagnosticForOptionName(Diagnostics.Option_0_cannot_be_specified_without_specifying_option_1, "declarationDir", "declaration");
+                if (!(options.declaration || options.composite)) {
```
Comment (ajafff) [inline src/compiler/program.ts]: there's a function `getEmitDeclarations` that does the same and is used in all other places

File: src/compiler/program.ts [hunk 2]
```diff
@@ -2532,8 +2532,8 @@ namespace ts {
             }
 
             if (options.declarationDir) {
-                if (!options.declaration) {
-                    createDiagnosticForOptionName(Diagnostics.Option_0_cannot_be_specified_without_specifying_option_1, "declarationDir", "declaration");
+                if (!(options.declaration || options.composite)) {
+                    createDiagnosticForOptionName(Diagnostics.Option_0_cannot_be_specified_with_option_1, "declarationDir", options.declaration ? "declaration" : "composite");
```
Comment (ajafff) [inline src/compiler/program.ts]: why use a different diagnostic? the new one seems wrong here

Comment (ajafff) [reply src/compiler/program.ts]: I guess you don't need to change the error message at all since `declaration` is implicitly enabled by `composite`

**Comment by a-tarasyuk** [comment]
@ajafff thanks, fixed.

**Pull request merged**